### PR TITLE
Feat/filter users

### PR DESCRIPTION
**Pull request by GnesisVSS**

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -68,9 +70,31 @@ export class UserController {
...
   )
-  @Patch(':rut')
-  update(@Param('rut') rut: string, @Body() updateUserDto: UpdateUserDto) {
-    return this.userService.update(rut, updateUserDto);
+  @Patch('update')
+  update(@Query('rut') rut: string, @Body() updateUserDto: UpdateUserDto) {
+    try {
+      return this.userService.update(rut, updateUserDto);
+    } catch (error) {
+      throw new BadRequestException("Error updating user")
```
Comment (LuchoSerruya) [inline src/user/user.controller.ts]: UserService already throws an exception if something goes wrong - you don't need to catch it and throw a new one in here

File: src/user/user.controller.ts [hunk 2]
```diff
@@ -68,9 +70,31 @@ export class UserController {
...
+  @Get('searchUsers')
+  async searchUsers(@Query() query: SearchUserDto) {
+    try {
+      const users = await this.userService.searchUsers(query);
+      return {
+        message: `Found ${users.length} users`,
+        data: { users },
+      };
+    } catch (error) {
+      throw new BadRequestException('Error searching users');
```
Comment (LuchoSerruya) [inline src/user/user.controller.ts]: UserService already throws an exception if something goes wrong - you don't need to catch it and throw a new one in here

**Review by LuchoSerruya — CHANGES_REQUESTED** [review]
Pending changes

**Comment by GnesisVSS** [comment]
Fixed requested changes.

**Review by LuchoSerruya — APPROVED** [review]

**Pull request merged**